### ISSUE TITLE
Several issues with brute-force protection

### ISSUE THREAD

**Issue by nighknight** [thorsten/phpMyFAQ#1333]
Since 2.9.8 there is a brute-force protection included in phpMyFAQ. Unfortunately, there are several issues with this protection forcing me to roll it back.

Effectively, if you type a wrong password once, you are locked out forever without a helpful error message. To get back access, you have to reset the "success" value in the database.

**Detailed description**
1. The brute-force protection is already activated after the first wrong password. This happens quite often as well for legitimate users (e.g. by mistyping or mixing up passwords). I would prefer a blocking only after maybe 5 failed attempts.

2. There is no helpful error message signaling the real reasons for subsequent authentication failures. The message shown is "Wrong login name or password":
![grafik](https://user-images.githubusercontent.com/29982026/28462084-ead7db1e-6e19-11e7-949e-f75bc42a1242.png)
An error message like "You exceeded the maximum amounts of login attempts and are temporarily blocked. Please try again later." would be helpful.

3. The unblocking does not work. Not only a failed attempt in the last 10 minutes is taken into account, but any last failed attempt: All entries with `session_timestamp < REQUEST_TIME + lockoutTime` are selected from the database.

See code from `phpmyfaq/inc/PMF/User/CurrentUser.php`:
```php
    protected function isFailedLastLoginAttempt()
    {
        $select = sprintf("
            SELECT
                session_timestamp,
                ip,
                success
            FROM
                %sfaquser
            WHERE
                user_id = %d
            AND
                session_timestamp < '%s'
            AND
                ip = '%s'
            AND
                success = 0",
            PMF_Db::getTablePrefix(),
            $this->getUserId(),
            $_SERVER['REQUEST_TIME'] + $this->lockoutTime,
            $_SERVER['REMOTE_ADDR']
        );

        $result = $this->config->getDb()->query($select);
        if ($this->config->getDb()->numRows($result) !== 0) {
            return true;
        } else {
            return false;
        }
    }
````

**Comment by thorsten** [comment]
Yes, I see some issues as well in the current implementation. Will work on it.

**Comment by thorsten** [comment]
1. can't be fixed in 2.9, we have to change the database layout for that, so this will be done in 2.10

**Comment by thorsten** [comment]
2. is now fixed

**Comment by thorsten** [comment]
3. is now fixed

**Comment by pekidi** [comment]
greetings

i have the same problem on new version 2.9.9. I am typing the correct password and faq does no accept it, otherwise, he locking my account. On database i see account as active.

Please help

regards
-D

**Comment by thorsten** [comment]
Are you sure it is locked?

**Comment by pekidi** [comment]
in database i see active account, but faq gives me info "you exceeded maximum amounts...."

**Comment by nighknight** [comment]
@thorsten 
As the issue is already closed, is point 1 (blocking only after a couple of wrong attempts) still on the list for 2.10?

**Comment by HankBrown** [comment]
This is cruel. Put the number of attempts much higher. Let us reduce it AFTER we have installed, played, tested and evaluated.  You are going to blow away the people who are just looking at the product. Hell if I know what my password is. It's  a dev/test machine, I can't send mail. I know what  my usual passwords and user names are, it will be one of them. Apparently I have more than you allow. It does not take long for combinations (name vs email + password) to add up.

Anyone else in pain here is a fast temporary workarround

/inc/PMF/User/CurrentUser.php

```
    public function login($login, $password)
    {
        $optData = [];
        $loginError = $passwordError = $count = 0;
/*
        // First check for brute force attack
        $this->getUserByLogin($login);
        if ($this->isFailedLastLoginAttempt()) {
            $this->errors[] = parent::ERROR_USER_TOO_MANY_FAILED_LOGINS;
            return false;
        }
*/

```

**Comment by pekidi** [comment]
working, thank u

**Comment by nighknight** [comment]
@thorsten: Are there currently any plans to block users only after like 5 failed logins (see my point 1)?

**Comment by thorsten** [comment]
@nighknight I could add it to 3.0. Please add a new issue if you want this in 3.0
